org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookActivity.java

```diff
@@ -157,6 +164,20 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
         }
     }
 
+    private Boolean isNetworkAvailable(Application application) {
+        ConnectivityManager connectivityManager = (ConnectivityManager)
+                application.getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            Network nw = connectivityManager.getActiveNetwork();
+            if (nw == null) return false;
+            NetworkCapabilities actNw = connectivityManager.getNetworkCapabilities(nw);
+            return true;
+        } else {
+            NetworkInfo nwInfo = connectivityManager.getActiveNetworkInfo();
+            return true;
+        }
+    }
+
```

Comment:
No need to implement a network check on your own, since the enviroCar _core _ module already contains a utility class for it: https://github.com/enviroCar/enviroCar-app/blob/master/org.envirocar.core/src/main/java/org/envirocar/core/ContextInternetAccessProvider.java

I'd suggest to use the existing class which makes it easier to react on any future API changes on this functionality, because you have to adopt the implementation only in one class.